### PR TITLE
classnames also accept undefined as value

### PR DESCRIPTION
**Pull request by pasix**
**What**:

**Why**:

**How**:

**Checklist**:

- [ ] Documentation
- [ ] Tests
- [ ] Code complete

### CODE REVIEW

File: packages/create-emotion/types/index.d.ts [hunk 1]
```diff
@@ -32,7 +32,7 @@ export interface ArrayClassNameArg extends Array<ClassNameArg> {}
 
 export type ClassNameArg =
   | undefined | null | boolean | string
-  | { [key: string]: boolean }
+  | { [key: string]: boolean | undefined }
```
Comment (tkh44) [inline packages/create-emotion/types/index.d.ts]: Would this need to support all of these types as values?

Comment (Ailrun) [reply packages/create-emotion/types/index.d.ts]: We may need `any`, as described in #746.

Comment (Andarist) [reply packages/create-emotion/types/index.d.ts]: while `any` is technically correct I doubt we should actually recommend it, typings give us power to restrict usages to sane ones

Comment (pasix) [reply packages/create-emotion/types/index.d.ts]: yes it's better with `undefined | null | boolean | string`

**Comment by osi-oswald** [comment]
Why was type `number` not added as well? Consider the following usecase:
```
cx('count', { isOdd: count % 2 }) // TS complains, emotion handles it just fine
```
I now have to type one of the following just to make TS happy again (which I find unnecessary):
```
cx('count', { Boolean(isOdd: count % 2) })
cx('count', { !!(isOdd: count % 2) })
```
Actually type `any` would really be the best type for it. Otherwise someone might come and ask "why did you not add type XXX as well"?

**Comment by Andarist** [comment]
Imho relaying on auto bool coercion for numbers there is not the best way to do. While it ofc works in JavaScript, type systems should strive to be more explicit about such things and I would consider the current lack of number support a good thing.

**Pull request merged**